### ISSUE TITLE
flip/rotate broken on latest versions

### ISSUE THREAD

**Issue by msx80** [nesbox/TIC-80#2016]
Hi there, the combination of flip and rotate is broken in 1.0.x. It was working on 0.90.
Affected games that i know of: Warm wheels, Secret Agents.

Test case:

```
function TIC()
 cls(0)
 for i=0,3 do
   for j=0,3 do
     spr(0,i*10,j*10,14,1,j,i)
   end
 end
end

```

On 0.9.something:

![image](https://user-images.githubusercontent.com/2278103/192991135-1852a53d-079c-45bb-874f-5a1bb561d0cf.png)

On 1.0.something:

![image](https://user-images.githubusercontent.com/2278103/192991186-5b9eb6f3-e8c2-4050-a410-0ce7d267de05.png)

**Comment by borbware** [comment]
I noticed the same recently. It's also broken on map()'s remap functions:

0.90:
![video21](https://user-images.githubusercontent.com/19384640/193931975-2bae9553-48d3-46d2-9079-2985a3ec604d.gif)

1.0:
![video20](https://user-images.githubusercontent.com/19384640/193931987-5fb310f4-ecf9-4049-9449-4cdb976d14e1.gif)

**Comment by borbware** [comment]
@nesbox any ideas what might have broken this?

**Comment by Gavin-Paolucci-Kleinow** [comment]
It will be great to have this bug fixes. L

**Comment by Podepi** [comment]
So the issue appears to be that someone wrote the wrong number when converting from binary to decimal on line 158 in `src/core/draw.c`
`orientation |= 0b100` became `orientation |= 2` instead of `orientation |= 4`
It's a simple fix, but I've never contributed to a git repository before so I'm not sure what to do. Do I just make a pull request with that single line changed?

**Comment by borbware** [comment]
So that's the issue! To contribute a fix, make a fork of the repo, change the line (and make a build so you can confirm the change fixes the issue), make a commit (mention "fixes #2016" or something), and then make a pull request where you also mention this issue #2016. That should do the trick!

**Comment by joshgoebel** [comment]
Yes someone whip up a PR please and I'm happy to approve.

Was this decimal conversion done for some very old compilers?  If so, we really probably should have a file full of just binary constants (or at least the small ones) to cover this instead of doing manual conversions.
